internal/buffer/settings.go

```diff
@@ -10,11 +10,9 @@ func (b *Buffer) SetOptionNative(option string, nativeValue interface{}) error {
 
 	if option == "fastdirty" {
 		if !nativeValue.(bool) {
-			if !b.Modified() {
-				e := calcHash(b, &b.origHash)
-				if e == ErrFileTooLarge {
-					b.Settings["fastdirty"] = false
-				}
+			e := calcHash(b, &b.origHash)
```

Comment:
Ehh... this causes a regression: modify the buffer (without saving it) and run `reload` -> statusbar now shows that the buffer is not modified (while it is) and Ctrl-Q closes the buffer without asking to save changes.

That's because we update `b.origHash` here, so `b.Modified()` now returns true. So now we know what was the `!b.Modified()` check for...

So for the case when the buffer is modified, we need to find out the buffer size in some other way.